apps/server/src/modules/learnroom/service/common-cartridge-import.service.ts

```diff
@@ -1,13 +1,42 @@
-import { CommonCartridgeFileParser } from '@modules/common-cartridge';
 import { Injectable } from '@nestjs/common';
+import { BoardExternalReferenceType } from '@shared/domain/domainobject';
 import { Course, User } from '@shared/domain/entity';
+import { ColumnBoardService, ColumnService } from '@src/modules/board';
+import { CommonCartridgeFileParser, DEFAULT_FILE_PARSER_OPTIONS } from '@src/modules/common-cartridge';
+import { CourseService } from './course.service';
 
 @Injectable()
 export class CommonCartridgeImportService {
-	public createCourse(user: User, file: Buffer): Course {
-		const parser = new CommonCartridgeFileParser(file);
+	constructor(
+		private readonly courseService: CourseService,
+		private readonly columnBoardService: ColumnBoardService,
+		private readonly columnService: ColumnService
+	) {}
+
+	public async importFile(user: User, file: Buffer): Promise<void> {
+		const parser = new CommonCartridgeFileParser(file, {
+			maxSearchDepth: 1,
+			pathSeparator: DEFAULT_FILE_PARSER_OPTIONS.pathSeparator,
+		});
 		const course = new Course({ teachers: [user], school: user.school, name: parser.manifest.getTitle() });
 
-		return course;
+		await this.courseService.create(course);
+		await this.createColumnBoard(course, parser);
+	}
+
+	private async createColumnBoard(course: Course, parser: CommonCartridgeFileParser): Promise<void> {
+		const organizations = parser.manifest.getOrganizations();
+		const columnBoard = await this.columnBoardService.create(
+			{
+				type: BoardExternalReferenceType.Course,
+				id: course.id,
+			},
+			parser.manifest.getTitle()
+		);
+		const columnProps = organizations.map((organization) => {
+			return { title: organization.title };
+		});
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		const columns = await this.columnService.createMany(columnBoard, columnProps);

```

Comment:
Why the constant here? Can't you just write 'await this.columnService.createMany(columnBoard, columnProps);' ?

Reply:
I thought I will use the return value in the future. I will remove it.

Reply:
The pipline does not show the these lines are not covered and the unit test calls implicit this functions.

Reply:
Done